os/kernel/binary_manager/binary_manager_getinfo.c

```diff
@@ -194,22 +201,36 @@ void binary_manager_get_info_with_name(int requester_pid, char *bin_name)
 	memset((void *)&response_msg, 0, sizeof(binmgr_getinfo_response_t));
 	response_msg.result = BINMGR_NOT_FOUND;
 
-	bin_count = binary_manager_get_ucount();
-	for (bin_idx = 1; bin_idx <= bin_count; bin_idx++) {
-		if (!strncmp(BIN_NAME(bin_idx), bin_name, BIN_NAME_MAX)) {
-			size = binary_manager_get_available_size(bin_idx);
-			if (size < 0) {
-				response_msg.result = BINMGR_OPERATION_FAIL;
-			} else {
-				response_msg.result = BINMGR_OK;
-				response_msg.data.available_size = size;
-				strncpy(response_msg.data.name, BIN_NAME(bin_idx) , BIN_NAME_MAX);
-				response_msg.data.version = (double)BIN_LOADVER(bin_idx);
+	if (!strncmp("kernel", bin_name, BIN_NAME_MAX)) {
+		kerinfo = binary_manager_get_kdata();
+		strncpy(response_msg.data.name, "kernel", BIN_NAME_MAX);
+		response_msg.data.version = kerinfo->version;
+		if (kerinfo->part_count > 1) {
+			response_msg.data.available_size = kerinfo->part_size[kerinfo->inuse_idx ^ 1];
+		} else {
+			response_msg.data.available_size = -1;
+		}
+		response_msg.result = BINMGR_OK;
+	}
+#ifdef CONFIG_APP_BINARY_SEPARATION
+	else {
+		bin_count = binary_manager_get_ucount();
+		for (bin_idx = 1; bin_idx <= bin_count; bin_idx++) {
+			if (!strncmp(BIN_NAME(bin_idx), bin_name, BIN_NAME_MAX)) {
+				size = binary_manager_get_available_size(bin_idx);
+				if (size < 0) {
```

Comment:
If the available size is 0, is it okay not to handle as fail?

Reply:
Yes. a size can be 0 on success(BINMGR_OK). It means operations are successful but a available space is not existing.